pkg/kritis/admission/admission.go

```diff
@@ -121,6 +139,24 @@ func handleReplicaSet(ar *v1beta1.AdmissionReview, admitResponse *v1beta1.Admiss
 		return err
 	}
 	glog.Infof("handling replica set %s...", replicaSet.Name)
+
+	operation := ar.Request.Operation
+	if operation == v1beta1.Update {
+		oldReplicaSet := appsv1.ReplicaSet{}
+		if err := json.Unmarshal(ar.Request.OldObject.Raw, &oldReplicaSet); err != nil {
+			return err
+		}
+
+		// For UPDATE events, if there is no new image added, we can skip the check.
+		// This is required, so that DELETE events work for ReplicaSet.
+		//
+		// Before deleting a replicaSet, kubernetes always make replicas to 0 which causes an
+		// UPDATE event.
+		if !hasNewImage(ReplicaSetImages(replicaSet), ReplicaSetImages(oldReplicaSet)) {
+			return nil
```

Comment:
Same as above, I'd like to add INFO logging here to know it's skipped because of the same image.